File: apps/api/src/main/java/com/lifttrack/api/domain/models/RoutineDay.java
package com.lifttrack.api.domain.models;

import java.time.Instant;
import java.util.UUID;

public record RoutineDay(
        UUID uuid,
        UUID routineUuid,
        int dayOfWeek,
        String name,
        int sortOrder,
        Instant createdAt
) {
}

File: apps/web/src/domain/models/routine-day.ts
export type DayOfWeek = 1 | 2 | 3 | 4 | 5 | 6 | 7

export interface RoutineDay {
  readonly uuid: string
  readonly routineUuid: string
  readonly dayOfWeek: DayOfWeek
  readonly name: string
  readonly sortOrder: number
  readonly createdAt: string
}

export interface CreateRoutineDay {
  readonly dayOfWeek: DayOfWeek
  readonly name: string
  readonly sortOrder: number
}
